htdocs/core/lib/pdf.lib.php

```diff
@@ -1172,17 +1185,35 @@ function pdf_pagefoot(&$pdf, $outputlangs, $paramfreetext, $fromcompany, $marge_
 			}
 
 			$pdf->SetY(-$posy);
-			$pdf->line($dims['lm'], $dims['hk'] - $posy, $dims['wk'] - $dims['rm'], $dims['hk'] - $posy);
-			$posy--;
 
-			$pdf->writeHTMLCell($pdf->page_largeur - $pdf->margin_left - $pdf->margin_right, $freetextheight, $dims['lm'], $dims['hk'] - $posy, dol_htmlentitiesbr($mycustomfooter, 1, 'UTF-8', 0));
+			// Hide footer line if footer background color is set
+			if (empty($conf->global->PDF_FOOTER_BACKGROUND_COLOR)) {
+				$pdf->line($dims['lm'], $dims['hk'] - $posy, $dims['wk'] - $dims['rm'], $dims['hk'] - $posy);
+			}
+
+			// Option for set top margin height of footer after freetext
+			if (!empty($conf->global->PDF_FOOTER_TOP_MARGIN) || ($conf->global->PDF_FOOTER_TOP_MARGIN === '0')) {
+				$posy -= $conf->global->PDF_FOOTER_TOP_MARGIN;
+			} else { $posy--; }
+
+			if ($conf->global->PDF_FOOTER_DISABLE_PAGEBREAK === '1') { $pdf->SetAutoPageBreak(0, 0); } // Option for disable auto pagebreak
+			$pdf->writeHTMLCell($pdf->page_largeur - $pdf->margin_left - $pdf->margin_right, $mycustomfooterheight, $dims['lm'], $dims['hk'] - $posy, dol_htmlentitiesbr($mycustomfooter, 1, 'UTF-8', 0));
+			$pdf->SetAutoPageBreak(1, 0); // Restore pagebreak
```

Comment:
Because you disabled pagebreak only 
`if ($conf->global->PDF_FOOTER_DISABLE_PAGEBREAK === '1') {`
you must alos enable it 
if ($conf->global->PDF_FOOTER_DISABLE_PAGEBREAK === '1') {

Reply:
Hello @eldy like this ?
Thank